PaxLabs/mynewt-core
# PKF-650: turn off I2C

1. Turn off I2C when we finish read/write
    2. Add mutex to prevent reentrant

Test:
This change is made to improve power consumption.
1. Make sure both pressure sensors work (e.g. psense stream -d, and verify the value)
2. Make sure accelerometer works (e.g. lis2dw12 r 5, and verify the value. Or, shake the device)
3. When system is in low power mode, the current should be less than 20uA.
